### PR TITLE
Add wake 0.18.1 typecheck CI step

### PR DESCRIPTION
**Pull request by mmjconolly**
Adds a CI test for running type checking of the wake rules against the newer wake version (0.18.1)

### CODE REVIEW

**Comment by mmjconolly** [comment]
We see 
```
$ wake -x Unit
Was expecting a EQUALS, but got a TYPE at 'type' (esdk-settings-generator/build.wake:24:[70-73])
Was expecting a EOL, but got a EQUALS at '=' (esdk-settings-generator/build.wake:24:86)
Was expecting a END, but got a EQUALS at '=' (esdk-settings-generator/build.wake:24:86)
Subscribe of 'driverImplementations' is to a non-existent topic at api-generator-sifive/dut/drivers.wake:30:[3-33]
Variable reference 'outputFile' is unbound at esdk-settings-generator/build.wake:24:[75-84]
Variable reference 'topic driverImplementations' is unbound at api-generator-sifive/dut/drivers.wake:30:[3-33]
Variable reference 'runESDKSettingsGenerator' is unbound at api-generator-sifive/program/freedom-metal-program.wake:22:[7-30]
```

**Comment by richardxia** [comment]
Re: the esdk-settings-generator errors, you just need to bump the wit manifest, since they were already fixed months ago.

Re: `topic driverImplementations`, we just need to define that topic here, probably just above the function that subscribes to it and just below the tuple definition of `DriverImplementation`.

I haven't see `runESDKSettingsGenerator`, even though I've been doing a lot of local bumps of repos to test out the wake 0.18 changes, so that might go away if everything is updated.

**Comment by mmjconolly** [comment]
Is it intentional that the tests are executed/displayed twice (push/pull_request) ?

**Comment by richardxia** [comment]
> Is it intentional that the tests are executed/displayed twice (push/pull_request) ?

Somewhat, but mostly out of laziness, and since the CI runs fast enough.

`push` is needed for post-merge CI, such as merging to master. Since `push` doesn't treat any particular branch as special, it also will run the CI on the feature branch you push to. You can specify individual branches that `push` should run on, but I didn't want to do that in case if we ever created a long-running release branch.

**Review by richardxia — APPROVED** [review]

File: dut/drivers.wake [hunk 1]
```diff
@@ -26,6 +26,8 @@ tuple DriverImplementation =
 global def makeDriverImplementation compatibleDevices vendor deviceName driverSources includeDirs visible =
   DriverImplementation compatibleDevices vendor deviceName driverSources includeDirs visible
 
+topic driverImplementations: DriverImplementation
```
Comment (richardxia) [inline dut/drivers.wake:29]: Whoops, should have reviewed this more closely. This should have the `global` keyword.

**Pull request merged**